pkg/icingadb/ha.go

```diff
@@ -428,9 +433,17 @@ func (h *HA) realize(
 
 		h.signalTakeover(takeover)
 	} else if otherResponsible {
-		if state, _ := h.state.Load(); !state.otherResponsible {
+		var state haState
+		statePtr := h.state.Load()
+		if statePtr != nil {
+			// Dereference pointer to create a copy of the value it points to.
+			// Ensures that any modifications do not directly affect the original data unless explicitly stored back.
+			state = *statePtr
+		}
+
+		if !state.otherResponsible {
```

Comment:
You don't change the state if you don't enter this `if` branch anyway, so I would handle this the same way as before.
```suggestion
		if state := h.state.Load(); state != nil && !state.otherResponsible {
			// Dereference pointer to create a copy of the value it points to.
			// Ensures that any modifications do not directly affect the original data unless explicitly stored back.
			newState := *state
			newState.otherResponsible = true
			h.state.Store(&newState)
		}
```

Reply:
Oh, never mind, I get it now why you did it that way 🙈